src/main/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJob.java

```diff
@@ -42,7 +42,9 @@ public void accept(JobContext ctx) throws Exception {
             for (UserCommons userCommons : allUserCommons) {
                 User user = userRepository.findById(userCommons.getUserId()).orElseThrow(() -> new RuntimeException("Error calling userRepository.findById(" + userCommons.getUserId() + ")"));
                 ctx.log("User: " + user.getFullName() + ", numCows: " + userCommons.getNumOfCows() + ", cowHealth: " + userCommons.getCowHealth());
-
+                
+                calculateCowDeaths(userCommons, ctx);
+                
                 var newCowHealth = calculateNewCowHealthUsingStrategy(cowHealthUpdateStrategy, commons, userCommons, totalCows);
                 ctx.log(" old cow health: " + userCommons.getCowHealth() + ", new cow health: " + newCowHealth);
                 userCommons.setCowHealth(newCowHealth);
```

Comment:
This function overrides resetting cow health to 100 because it is using the old cow number of cows `totalCows`. I think it should be ok to replace it with `userCommons.totalCows` (on line 65).